### PR TITLE
Add iframe dispatcher

### PR DESCRIPTION
**Pull request by eluxen**
@liorgoldhamer @rauchy @yairperman @yuval-netanel @arbeli @hagail @talyas 
@tomerlevi444

To test it
1) use https://stash.internal.machines/users/lior.goldhamer/repos/sinatra-with-jockey/browse as a demo web client
2) run inapp from the test-iframe-dispatcher branch
3) run `ngrok --subdomain lux 3000`

### CODE REVIEW

File: JockeyJS/js/jockey.js [hunk 1]
```diff
@@ -150,7 +186,29 @@
...
+        activateIframeDispatcher: function(targetDomain, targetWindow) {
+            this.targetDomain = targetDomain || '*'
+            this.targetWindow = targetWindow || window.parent
+            window.addEventListener("message", this.onMessageRecieved.bind(this), false);
+            this.dispatchers.push(IframeDispatcher);
+        },
+
+        // Handles postMessage events when iframeDispatcher is used
+        onMessageRecieved: function(event) {
+            if (this.targetDomain != '*' && this.targetDomain != event.origin)
```
Comment (yuval-netanel) [inline JockeyJS/js/jockey.js]: Use curly brackets (it's not coffee)

File: JockeyJS/js/jockey.js [hunk 2]
```diff
@@ -150,7 +186,29 @@
...
+                return;
+
+            var envelope = event.data.envelope;
```
Comment (yuval-netanel) [inline JockeyJS/js/jockey.js:209]: Use curly brackets (it's not coffee)

**Comment by yuval-netanel** [comment]
Add documentation to read.me

File: JockeyJS/js/jockey.js [hunk 3]
```diff
@@ -187,14 +238,12 @@
...
-	
-    if ((iOS && UIWebView) || isAndroid) { 
-        Jockey.dispatcher = Dispatcher;
-    } else {
-        Jockey.dispatcher = nullDispatcher;
+    var UIWebView  = /(iPhone|iPod|iPad).*AppleWebKit(?!.*Safari)/i.test(navigator.userAgent);
+    var isAndroid  = navigator.userAgent.toLowerCase().indexOf("android") > -1;
+
+    if ((iOS && UIWebView) || isAndroid) {
+        Jockey.dispatchers.push(nativeDispatcher);
```
Comment (yuval-netanel) [inline JockeyJS/js/jockey.js:248]: nativeDispatcher + iframeDispatcher won't work on iphone (which don't have the native jockey lib)

File: JockeyJS/js/jockey.js [hunk 4]
```diff
@@ -150,7 +186,29 @@
...
         // a given message.
         triggerCallback: function(id) {
-            this.dispatcher.triggerCallback(id);
+            this.dispatchers.forEach(function(dispatcher) {
+                dispatcher.triggerCallback(id);
+            });
+        },
+
+        activateIframeDispatcher: function(targetDomain, targetWindow) {
+            this.targetDomain = targetDomain || '*'
```
Comment (yuval-netanel) [inline JockeyJS/js/jockey.js]: ;

File: JockeyJS/js/jockey.js [hunk 5]
```diff
@@ -150,7 +186,29 @@
...
+            this.targetWindow = targetWindow || window.parent
```
Comment (yuval-netanel) [inline JockeyJS/js/jockey.js]: ;

File: JockeyJS/js/jockey.js (hunk 1 above)
Comment (eluxen) [inline JockeyJS/js/jockey.js]: done

File: JockeyJS/js/jockey.js (hunk 2 above)
Comment (eluxen) [inline JockeyJS/js/jockey.js:209]: done

File: JockeyJS/js/jockey.js (hunk 4 above)
Comment (eluxen) [inline JockeyJS/js/jockey.js]: done

File: JockeyJS/js/jockey.js (hunk 5 above)
Comment (eluxen) [inline JockeyJS/js/jockey.js]: done

File: JockeyJS/js/jockey.js (hunk 3 above)
Comment (eluxen) [inline JockeyJS/js/jockey.js:248]: done

**Comment by eluxen** [comment]
How do you think we should add tests here? The project doesn't have js tests, should we add some kind of an iframe example?

**Comment by arbeli** [comment]
+1

**Pull request closed without merging**